script/tool/lib/src/version_check_command.dart

```diff
@@ -100,175 +104,175 @@ class VersionCheckCommand extends PluginCommand {
+  Future<void> initializeRun() async {}
 
-    final List<String> changedPubspecs =
-        await gitVersionFinder.getChangedPubSpecs();
+  @override
+  Future<List<String>> runForPackage(Directory package) async {
+    final List<String> errors = <String>[];
 
-    final List<String> badVersionChangePubspecs = <String>[];
+    final Pubspec? pubspec = _tryParsePubspec(package);
+    Pubspec.parse(package.childFile('pubspec.yaml').readAsStringSync());
+    if (pubspec == null) {
+      errors.add('Invalid pubspec.yaml.');
+      return errors; // No remaining checks make sense.
+    }
 
-    const String indentation = '  ';
-    for (final String pubspecPath in changedPubspecs) {
-      print('Checking versions for $pubspecPath...');
-      final File pubspecFile = packagesDir.fileSystem.file(pubspecPath);
-      if (!pubspecFile.existsSync()) {
-        print('${indentation}Deleted; skipping.');
-        continue;
-      }
-      final Pubspec pubspec = Pubspec.parse(pubspecFile.readAsStringSync());
-      if (pubspec.publishTo == 'none') {
-        print('${indentation}Found "publish_to: none"; skipping.');
-        continue;
-      }
+    if (pubspec.publishTo == 'none') {
+      printSkip('${indentation}Found "publish_to: none".');
+      return PackageLoopingCommand.success;
+    }
 
-      final Version? headVersion =
-          await gitVersionFinder.getPackageVersion(pubspecPath, gitRef: 'HEAD');
-      if (headVersion == null) {
-        printError('${indentation}No version found. A package that '
-            'intentionally has no version should be marked '
-            '"publish_to: none".');
-        badVersionChangePubspecs.add(pubspecPath);
-        continue;
-      }
-      Version? sourceVersion;
-      if (getBoolArg(_againstPubFlag)) {
-        final String packageName = pubspecFile.parent.basename;
-        final PubVersionFinderResponse pubVersionFinderResponse =
-            await _pubVersionFinder.getPackageVersion(package: packageName);
-        switch (pubVersionFinderResponse.result) {
-          case PubVersionFinderResult.success:
-            sourceVersion = pubVersionFinderResponse.versions.first;
-            print(
-                '$indentation$packageName: Current largest version on pub: $sourceVersion');
-            break;
-          case PubVersionFinderResult.fail:
-            printError('''
+    final Version? currentPubspecVersion = pubspec.version;
+    if (currentPubspecVersion == null) {
+      printError('${indentation}No version found in pubspec.yaml. A package '
+          'that intentionally has no version should be marked '
+          '"publish_to: none".');
+      errors.add('No pubspec.yaml version.');
+      return errors; // No remaining checks make sense.
+    }
+
+    if (!await _checkVersionChange(package, pubspec: pubspec)) {
+      errors.add('Disallowed version change.');
+    }
+
+    if (!(await _checkVersionsMatch(package, pubspec: pubspec))) {
```

Comment:
Same comment about "check", how about `_doVersionsMatch`?

Reply:
Changed to _hasConsistentVersion (for similar reasons; the arguments are not version being compared).